SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SchemaGenerator.cpp

```diff
@@ -313,13 +314,15 @@ int GenerateActorSchema(int ComponentId, UClass* Class, TSharedPtr<FUnrealType>
 			continue;
 		}
 
+		const Worker_ComponentId CachedComponentId = SchemaData ? SchemaData->SchemaComponents[PropertyGroupToSchemaComponentType(Group)] : 0;
+
 		Writer.PrintNewLine();
 
 		Writer.Printf("component {0} {", *SchemaReplicatedDataName(Group, Class));
 		Writer.Indent();
-		Writer.Printf("id = {0};", IdGenerator.GetNextAvailableId());
+		Writer.Printf("id = {0};", CachedComponentId > 0 ? CachedComponentId : IdGenerator.GetNextAvailableId());
```

Comment:
The pattern of `CachedComponentId > 0 ? CachedComponentId : IdGenerator.GetNextAvailableId()` appears about 8 times. Could this be condensed maybe? Maybe making the IdGenerator knowledgeable about cached component ids?